runtime/compiler/build/toolcfg/gnu/common.mk

```diff
@@ -532,30 +528,25 @@ ifneq ($(JITSERVER_SUPPORT),)
     SOLINK_SLINK_STATIC=-l:libprotobuf.a
     CXX_DEFINES+=GOOGLE_PROTOBUF_NO_RTTI
 
-    ifneq ($(JITSERVER_ENABLE_SSL),)
-        SOLINK_SLINK+=ssl
-
-        ifneq ($(OPENSSL_CFLAGS),)
-            ADD_OPENSSL_CFLAGS=false
-            # --with-openssl=system, OPENSSL_LIBS looks like "-L/path/to/system/openssllib -lssl -lcrypto"
-            ifneq ($(OPENSSL_LIBS),)
-                SOLINK_LIBPATH+=$(subst -L,,$(firstword $(OPENSSL_LIBS)))
-                ADD_OPENSSL_CFLAGS=true
-            # --with-openssl=fetched --enable-openssl-bundling
-            else ifneq ($(OPENSSL_BUNDLE_LIB_PATH),)
-                SOLINK_LIBPATH+=$(OPENSSL_BUNDLE_LIB_PATH)
-                SOLINK_FLAGS+=-Wl,-rpath,\$$ORIGIN/..
-                ADD_OPENSSL_CFLAGS=true
-            # --with-openssl=fetched only
-            else ifneq ($(OPENSSL_DIR),)
-                SOLINK_LIBPATH+=$(OPENSSL_DIR)
-                ADD_OPENSSL_CFLAGS=true
-            endif
-
-            ifeq (true, $(ADD_OPENSSL_CFLAGS))
-                C_INCLUDES+=$(subst -I,,$(OPENSSL_CFLAGS))
-                CXX_INCLUDES+=$(subst -I,,$(OPENSSL_CFLAGS))
-            endif
-        endif # OPENSSL_CFLAGS
-    endif # JITSERVER_ENABLE_SSL
+    SOLINK_SLINK+=ssl
+
+    ifneq ($(OPENSSL_CFLAGS),)
+        ADD_OPENSSL_CFLAGS=false
+        # --with-openssl=fetched --enable-openssl-bundling
+        ifneq ($(OPENSSL_BUNDLE_LIB_PATH),)
+            SOLINK_SLINK+=crypto
+            SOLINK_LIBPATH+=$(OPENSSL_BUNDLE_LIB_PATH)
+            SOLINK_FLAGS+=-Wl,-rpath,\$$ORIGIN/..
+            ADD_OPENSSL_CFLAGS=true
+        # --with-openssl=fetched only
+        else ifneq ($(OPENSSL_DIR),)
+            SOLINK_LIBPATH+=$(OPENSSL_DIR)
+            ADD_OPENSSL_CFLAGS=true
+        endif
+
+        ifeq (true, $(ADD_OPENSSL_CFLAGS))
+            C_INCLUDES+=$(subst -I,,$(OPENSSL_CFLAGS))
+            CXX_INCLUDES+=$(subst -I,,$(OPENSSL_CFLAGS))
```

Comment:
Shouldn't `$(OPENSSL_CFLAGS)` instead be added (unmodified) to `CX_FLAGS`?

Reply:
All other included paths are set up in `C_INCLUDES` and `CXX_INCLUDES`. It makes more sense to add libssl include path here too. 

```
for example:
$C_INCLUDES is [/root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/include_core /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/thread /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/gc/include /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/x/amd64 /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/x /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/x/amd64 /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/x /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/.. /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../codert_vm /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../gc_include /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../include /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../gc_glue_java /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../jit_vm /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../nls /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../oti /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../util /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/net/gen /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/net /root/home/ahuo/src/openj9-openjdk-jdk8/openssl/include]

$CXX_INCLUDES is [/root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/include_core /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/thread /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/gc/include /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/x/amd64 /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/x /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/x/amd64 /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/x /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/.. /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../codert_vm /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../gc_include /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../include /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../gc_glue_java /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../jit_vm /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../nls /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../oti /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../util /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/net/gen /root/home/ahuo/src/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/net /root/home/ahuo/src/openj9-openjdk-jdk8/openssl/include]
```

Reply:
The name `OPENSSL_CFLAGS` suggests that it may be more than just include path options.

Reply:
Worse the substitution means I can't refer to a version of openssl at `.../Some-Interim-Version/...` as that `-I` will be removed to get `.../Somenterim-Version/...`.

Reply:
Will add `OPENSSL_CFLAGS` to `C_FLAGS` and `CXX_FLAGS`. `CX_FLAGS` has been assigned to `C_FLAGS` and `CXX_FLAGS` way before` JITServer` related stuff is processed here.